mne/annotations.py

```diff
@@ -701,9 +701,48 @@ def _ensure_annotation_object(obj):
                          'mne.Annotations. Got %s.' % obj)
 
 
+def _select_annotations_based_on_description(descriptions, event_id=None,
+                                             regexp=None):
+    """Get a collection of descriptions and returns index of selected."""
+
+    regexp_comp = re.compile(regexp)
```

Comment:
this is weird. You have None as default but you cannot call re.compile with None.